test/src/test/java/org/axonframework/test/fixture/AxonTestFixtureMessagingTest.java

```diff
@@ -0,0 +1,274 @@
+               .events(
+                       studentNameChangedEvent("my-studentId-1", "name-1", 1),
+                       studentNameChangedEvent("my-studentId-1", "name-2", 1)
+               ).and()
+               .when()
+               .command(new ChangeStudentNameCommand("my-studentId-1", "name-3"))
+               .then()
+               .success()
+               .resultMessagePayload(new CommandResult("Result name-3"));
+    }
+
+    @Test
+    void givenEventWhenCommandThenExpectEvents() {
+        var configurer = MessagingConfigurer.create();
+        var studentEvents = new ArrayList<>();
+        configurer.registerEventSink(c -> (events) -> {
+            studentEvents.addAll(events);
+            return CompletableFuture.completedFuture(null);
+        });
+        configurer.registerCommandBus(
+                c -> new SimpleCommandBus()
+                        .subscribe(
+                                new QualifiedName(ChangeStudentNameCommand.class),
+                                (command, context) -> {
+                                    ChangeStudentNameCommand payload = (ChangeStudentNameCommand) command.getPayload();
+                                    var eventSink = c.getComponent(EventSink.class);
+                                    var changeNo = studentEvents.size() + 1;
+                                    eventSink.publish(context,
+                                                      studentNameChangedEventMessage(payload.id(), payload.name(),
+                                                                                     changeNo));
+                                    return MessageStream.empty().cast();
+                                }));
+
+        var fixture = AxonTestFixture.with(configurer);
+
+        fixture.given()
+               .event(studentNameChangedEventMessage("my-studentId-1", "name-1", 1))
+               .when()
+               .command(new ChangeStudentNameCommand("my-studentId-1", "name-1"))
+               .then()
+               .events(studentNameChangedEventMessage("my-studentId-1", "name-1", 2));
+    }
+
+    @Test
+    void whenEventThenExpectedCommand() {
+        var configurer = MessagingConfigurer.create();
+        AtomicBoolean eventHandled = new AtomicBoolean(false);
+        registerChangeStudentNameHandlerReturnsSingle(configurer);
+        configurer.registerEventSink(c -> (events) -> {
+            if (!eventHandled.getAndSet(true)) {
+                var commandGateway = c.getComponent(CommandGateway.class);
+                commandGateway.sendAndWait(new ChangeStudentNameCommand("id", "name"));
+            }
+            return CompletableFuture.completedFuture(null);
+        });
+
+        var fixture = AxonTestFixture.with(configurer);
+
+        fixture.when()
+               .event(new StudentNameChangedEvent("my-studentId-1", "name-1", 1))
+               .then()
+               .commands(new ChangeStudentNameCommand("id", "name"));
```

Comment:
Now you can test "event handlers" which cause commands!

Reply:
Loving it :-)